### ISSUE TITLE
Custom credential not setting environment variables on remote host when used along with machine credential

### ISSUE THREAD

**Issue by Uttam92477** [ansible/awx#5452]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
On Tower, when using a credential of type 'Custom Credential type' along with credential of type 'Machine credential', the custom credential is not setting environment variables on remote host. I'm getting 'None' when trying to access the said environment variables using env_fallback in module params in a custom module.
**Note:** The same thing is working if I'm not using 'machine_credential' and using only the 'custom_credential' and executing a command on 'localhost'

##### ENVIRONMENT
* AWX version: 3.5.2
* AWX install method: standalone on RHEL 7
* Ansible version:  2.8
* Operating System: Linux
* Web Browser: Chrome

##### STEPS TO REPRODUCE

- Create a custom ansible module 'custom_module' that executes a 'Oracle dataguard dgmgrl command' using os.system() . 
```
module_args = dict(
        database_username=dict(type='str',
                               required=False,
                               fallback=(env_fallback, ['ORACLE_DATABASE_USERNAME'])
                               ),
        database_password=dict(type='str',
                               required=False,
                               no_log=True,
                               fallback=(env_fallback, ['ORACLE_DATABASE_PASSWORD'])
                               ),
    )
```
- In the 'custom_module' module_args dict, create an env_fallback for oracle database username and password
- Create a custom credential type 'test_credential_type' in Tower 
**Input configuration of Custom credential type:**
```
fields:
  - id: username
    type: string
    label: Username
  - id: password
    type: string
    label: Password
    secret: true
required:
  - username
  - password
```
**Injector configuration of Custom credential type:**
```
env:
  ORACLE_DATABASE_PASSWORD: '{{ password }}'
  ORACLE_DATABASE_USERNAME: '{{ username }}'
```
- Create a credential 'test_credential' of type 'test_credential_type'
- Try to assign 'database_username' to result
`result['test_username_value']=module.params['database_username']
- Exit module 'custom_module'
- Create a playbook which executes the 'custom_module' on a remote host (Oracle linux)
   - In the playbook, you login using machine credentail from tower, which escalates to root, then to user 'oracle' using 
```
become: yes
become_user: oracle
```

##### EXPECTED RESULTS

I expect the result dict to contain the username passed in the 'custom_credential'

##### ACTUAL RESULTS

the result['test_usernam'] is **None** or empty

**Comment by Uttam92477** [comment]
Probably not a bug. It might have something to do with the way things are executed by Ansible. I raised the bug without knowing about it.  All the modules are analyzed/ partially executed in the 'Ansible controller' and only necessary communication is sent to remote host/ managed host via ssh or other methods. I wrote the custom module assuming it will be executed **on** remote host which I'm executing the module **against**

Do correct me if I'm wrong.
